src/transformers/models/cpmant/tokenization_cpmant.py

```diff
@@ -0,0 +1,271 @@
+    return vocab
+
+
+class WordpieceTokenizer(object):
+    def __init__(self, vocab, unk_token="<unk>", max_input_chars_per_word=200):
+        self.vocab = vocab
+        self.unk_token = unk_token
+        self.max_input_chars_per_word = max_input_chars_per_word
+
+    def tokenize(self, token):
+        chars = list(token)
+        if len(chars) > self.max_input_chars_per_word:
+            return [self.unk_token]
+
+        start = 0
+        sub_tokens = []
+        while start < len(chars):
+            end = len(chars)
+            cur_substr = None
+            while start < end:
+                substr = "".join(chars[start:end])
+                if substr in self.vocab:
+                    cur_substr = substr
+                    break
+                end -= 1
+            if cur_substr is None:
+                sub_tokens.append(self.unk_token)
+                start += 1
+            else:
+                sub_tokens.append(cur_substr)
+                start = end
+
+        return sub_tokens
+
+
+class CPMAntTokenizer(PreTrainedTokenizer):
+    """
+    Construct a CPMAnt tokenizer. Based on byte-level Byte-Pair-Encoding.
+
+    Args:
+        vocab_file (`str`):
+            Path to the vocabulary file.
+        bod_token (`str`, *optional*, defaults to `<d>`):
+            The beginning of document token.
+        eod_token (`str`, *optional*, defaults to `</d>`):
+            The end of document token.
+        bos_token (`str`, *optional*, defaults to `<s>`):
+            The beginning of sequence token.
+        eos_token (`str`, *optional*, defaults to `</s>`):
+            The end of sequence token.
+        pad_token (`str`, *optional*, defaults to `<pad>`):
+            The token used for padding.
+        unk_token (`str`, *optional*, defaults to `<unk>`):
+            The unknown token.
+        line_token (`str`, *optional*, defaults to `</n>`):
+            The line token.
+        space_token (`str`, *optional*, defaults to `</_>`):
+            The space token.
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    model_input_names = ["input_ids", "attention_mask"]
+
+    def __init__(
+        self,
+        vocab_file,
+        bod_token="<d>",
+        eod_token="</d>",
+        bos_token="<s>",
+        eos_token="</s>",
+        pad_token="<pad>",
+        unk_token="<unk>",
+        line_token="</n>",
+        space_token="</_>",
+        **kwargs,
+    ):
+        super().__init__(
+            bod_token=bod_token,
+            eod_token=eod_token,
+            bos_token=bos_token,
+            eos_token=eos_token,
+            pad_token=pad_token,
+            unk_token=unk_token,
+            line_token=line_token,
+            space_token=space_token,
+            padding_side="left",
+            **kwargs,
+        )
+        self.bod_token = bod_token
```

Comment:
There should be a `requires_backends["jieba"]` here, so that there is an error early if the dep is not installed.

Reply:
Added. `requires_backend(self, ["jieba])`